libraries/pbrlib/genglsl/lib/mx_microfacet_specular.glsl

```diff
@@ -168,3 +168,219 @@ vec3 mx_fresnel_conductor(float cosTheta, vec3 n, vec3 k)
+        return;
+    }
+
+    float cosTheta_t = sqrt(1 - eta2 * st2);
+    vec2 r = vec2((n2*cosTheta - n1*cosTheta_t) / (n2*cosTheta + n1*cosTheta_t),
+                  (n1*cosTheta - n2*cosTheta_t) / (n1*cosTheta + n2*cosTheta_t));
+    F = mx_square(r);
+    phi.x = (r.x < 0.0) ? M_PI : 0.0;
+    phi.y = (r.y < 0.0) ? M_PI : 0.0;
+}
+
+// Fresnel for dielectric/conductor interface and polarized light.
+// TODO: Optimize this functions and support wavelength dependent complex refraction index.
+void mx_fresnel_conductor_polarized(float cosTheta, float n1, float n2, float k, out vec2 F, out vec2 phi)
+{
+    if (k == 0)
+    {
+        // Use dielectric formula to avoid numerical issues
+        mx_fresnel_dielectric_polarized(cosTheta, n1, n2, F, phi);
+        return;
+    }
+
+    float A = mx_square(n2) * (1.0 - mx_square(k)) - mx_square(n1) * (1.0 - mx_square(cosTheta));
+    float B = sqrt(mx_square(A) + mx_square(2.0 * mx_square(n2) * k));
+    float U = sqrt((A+B) / 2.0);
+    float V = sqrt((B-A) / 2.0);
+
+    F.y = (mx_square(n1*cosTheta - U) + mx_square(V)) / (mx_square(n1*cosTheta + U) + mx_square(V));
+    phi.y = atan(2.0*n1 * V*cosTheta, mx_square(U) + mx_square(V) - mx_square(n1*cosTheta)) + M_PI;
+
+    F.x = (mx_square(mx_square(n2) * (1.0 - mx_square(k)) * cosTheta - n1*U) + mx_square(2.0 * mx_square(n2) * k * cosTheta - n1*V)) /
+            (mx_square(mx_square(n2) * (1.0 - mx_square(k)) * cosTheta + n1*U) + mx_square(2.0 * mx_square(n2) * k * cosTheta + n1*V));
+    phi.x = atan(2.0 * n1 * mx_square(n2) * cosTheta * (2*k*U - (1.0 - mx_square(k)) * V), mx_square(mx_square(n2) * (1.0 + mx_square(k)) * cosTheta) - mx_square(n1) * (mx_square(U) + mx_square(V)));
+}
+
+// XYZ to CIE 1931 RGB color space (using neutral E illuminant)
+const mat3 XYZ_TO_RGB = mat3(2.3706743, -0.5138850, 0.0052982, -0.9000405, 1.4253036, -0.0146949, -0.4706338, 0.0885814, 1.0093968);
+
+// Depolarization functions for natural light
+float mx_depolarize(vec2 v)
+{
+    return 0.5 * (v.x + v.y);
+}
+vec3 mx_depolarize(vec3 s, vec3 p)
+{
+    return 0.5 * (s + p);
+}
+
+// Evaluation XYZ sensitivity curves in Fourier space
+vec3 mx_eval_sensitivity(float opd, float shift)
+{
+    // Use Gaussian fits, given by 3 parameters: val, pos and var
+    float phase = 2*M_PI * opd;
+    vec3 val = vec3(5.4856e-13, 4.4201e-13, 5.2481e-13);
+    vec3 pos = vec3(1.6810e+06, 1.7953e+06, 2.2084e+06);
+    vec3 var = vec3(4.3278e+09, 9.3046e+09, 6.6121e+09);
+    vec3 xyz = val * sqrt(2*M_PI * var) * cos(pos * phase + shift) * exp(- var * phase*phase);
+    xyz.x   += 9.7470e-14 * sqrt(2*M_PI * 4.5282e+09) * cos(2.2399e+06 * phase + shift) * exp(- 4.5282e+09 * phase*phase);
+    return xyz / 1.0685e-7;
+}
+
+// A Practical Extension to Microfacet Theory for the Modeling of Varying Iridescence
+// https://belcour.github.io/blog/research/2017/05/01/brdf-thin-film.html
+vec3 mx_fresnel_airy(float cosTheta, vec3 ior, vec3 extinction, float tf_thickness, float tf_ior)
```

Comment:
New Fresnel model with Airy reflectance, from this paper, supporting thin film iridescence effects.